drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/binary/downloading/source/GitHubSource.java

```diff
@@ -101,13 +94,15 @@ protected ZonedDateTime extractModificationDate(HttpClient.Response response) {
     public ExternalBinary getReleaseForVersion(String version) throws Exception {
         String url = projectUrl + RELEASES_URL;
         int pageNumber = 1;
+        List<String> availableVersions = new ArrayList<>();
 
         while (true) {
             HttpClient.Response response = sentGetRequestWithPagination(url, pageNumber++, Collections.emptyMap());
             JsonElement releases = gson.fromJson(response.getPayload(), JsonElement.class);
 
             if (releases != null && releases.isJsonArray() && releases.getAsJsonArray().size() > 0) {
-                ExternalBinary releaseForVersion = getReleaseForVersion(version, releases.getAsJsonArray());
+                ExternalBinary releaseForVersion =
```

Comment:
Yes, something like this should work too:
~~~java
 @Override
    public ExternalBinary getReleaseForVersion(String version) throws Exception {
        String url = projectUrl + RELEASES_URL;
        int pageNumber = 1;
        List<String> availableVersions = new ArrayList<>();
        JsonElement releases = getReleases(url, pageNumber);

        while (releases != null && releases.isJsonArray() && releases.getAsJsonArray().size() > 0) {

            ExternalBinary releaseForVersion =
                getReleaseForVersion(version, releases.getAsJsonArray(), availableVersions);

            if (releaseForVersion != null) {
                return releaseForVersion;
            }
            releases = getReleases(url, ++pageNumber);
        }

        throw new IllegalArgumentException(
            "There wasn't found any release for the version: " + version + " in the repository: " + projectUrl
                + " Available versions are: " + availableVersions);
    }

    private JsonElement getReleases(String url, int pageNumber) throws Exception {
        HttpClient.Response response = sentGetRequestWithPagination(url, pageNumber, Collections.emptyMap());
        return gson.fromJson(response.getPayload(), JsonElement.class);
    }
~~~